### ISSUE TITLE
Reactive dict/map between documents

### ISSUE THREAD

**Issue by mitar** [peerlibrary/meteor-peerdb#9]
It seems a common data structure one would like is the following:

```
{
  related: {
    <id of a document>: {
      subfield: "Subfield value"
    },
    <id of a document>: {
      ...
    }
  }
}
```

So a dict/map where keys are IDs and values are a subdocument with few fields of a referenced document. One can do that as a list for now, but I think having a dict/map would be beneficial for easy direct access if key (document id) is known. But current code does not yet support keeping subdocuments in sync for such structure.

**Comment by moonrockfamily** [comment]
I'm interested in your use case for identified related documents with a subset of fields kept sync'd.  Can you share your example/general case.  Much appreciated.

**Comment by mitar** [comment]
I do not understand your question. This ticket is about one particular structure of data and not a general ticket for questions about PeerDB.

**Comment by moonrockfamily** [comment]
It seems I misunderstood your idea/proposal.  I initially thought you were proposing an alternative/extension to the existing ReferenceField capability fieldName: @ReferenceField and was seeking to understand your use case/why it wasn't sufficient... as per [README](https://github.com/peerlibrary/meteor-peerdb): 
```coffeescript
class Post extends Document
  # Other fields:
  #   body

  @Meta
    name: 'Post'
    fields: =>
      # We can reference other document
      author: @ReferenceField Person, ['username', 'displayName']
      # Or an array of documents
      subscribers: [@ReferenceField Person]
      reviewers: [@ReferenceField Person, ['username', 'displayName']]
```

**Comment by mitar** [comment]
Yes. Here is a ticket for a different structure than what is currently implemented. But I have not seen really a need for it. I prefer to use list of subdocuments anyway. Also, I do not like data schemas where keys have meaning. So ticket/idea is here, but this is it.

**Comment by mitar** [comment]
This can also work only if `<id of a document>` does not contain a dot.
